Legitimate failure:

```
13:33:04  Failed to compile test:
13:33:04  
13:33:04  # github.com/cilium/cilium/test/k8sT
13:33:04  k8sT/Policies.go:41:9: undefined: helpers.RunsOnGKE
```

You can test this locally with `make -C test build`.

--- a/Documentation/check-crd-compat-table.sh
+++ b/Documentation/check-crd-compat-table.sh
@@ -25,7 +25,7 @@ get_line_of_schema_version(){
 
 get_schema_of_branch(){
    stable_branch="${1}"
-   git grep -o 'CustomResourceDefinitionSchemaVersion =.*' ${remote}/${stable_branch} -- pkg/k8s | sed 's/.*=\ "//;s/"//'
+   git grep -o 'CustomResourceDefinitionSchemaVersion =.*' ${remote}/${stable_branch} -- pkg/k8s | sed 's/.*=\ "//;s/"//' | uniq
 }
 
 get_stable_branches(){

--- a/Documentation/check-crd-compat-table.sh
+++ b/Documentation/check-crd-compat-table.sh
@@ -34,10 +34,9 @@ get_stable_branches(){
 
 get_stable_tags_for_minor(){
    minor_ver="${1}"
-   git ls-remote --tags ${remote} v\* \
+   git ls-remote --refs --tags ${remote} v\* \
        | awk '{ print $2 }' \
        | grep "${minor_ver}" \
-       | grep -v '\^' \
        | grep -v '\-' \
        | sed 's+refs/tags/++' \
        | sort -V \
@@ -46,7 +45,11 @@ get_stable_tags_for_minor(){
 
 get_rc_tags_for_minor(){
    minor_ver="${1}"
-   git ls-remote --tags ${remote} v\* | awk '{ print $2 }' | grep "${minor_ver}" | grep -v '\^' | grep '\-' | sed 's+refs/tags/++' | sort -V
+   git ls-remote --refs --tags ${remote} v\* \
+       | awk '{ print $2 }' \
+       | grep "${minor_ver}.*\-" \
+       | sed 's+refs/tags/++' \
+       | sort -V
 }
 
 filter_out_oldest() {
@@ -144,16 +147,13 @@ release_version="v$release_ersion"
 
 create_file ${release_version} "${dst_file}"
 
-last_cilium_release=$(egrep "[ ]${release_version}[ ]" -B2 "${dst_file}" | head -n 1 | awk '{ print $2 }')
-last_release_version=$(egrep "[ ]${release_version}[ ]" -B2 "${dst_file}" | head -n 1 | awk '{ print $4 }')
-current_release_version=$(egrep "[ ]${release_version}[ ]" "${dst_file}" | head -n 1 | awk '{ print $4 }')
+last_cilium_release=$(egrep "[ ]${release_version}[ ]" -B2 "${dst_file}" | awk 'NR == 1 { print $2 }')
+last_release_version=$(egrep "[ ]${release_version}[ ]" -B2 "${dst_file}" | awk 'NR == 1 { print $4 }')
+current_release_version=$(egrep "[ ]${release_version}[ ]" "${dst_file}" | awk 'NR == 1 { print $4 }')
 >&2 echo "Cilium ${last_cilium_release} schema: ${last_release_version}; Current: ${current_release_version}"
 
 # Cilium v1.9 or earlier used examples/crds, this dir was moved in v1.10.
-crd_path="examples/crds"
-if [[ -L $crd_path && -d $crd_path ]]; then
-    crd_path="pkg/k8s/apis/cilium.io/client/crds"
-fi
+crd_path="$(realpath --relative-to . "examples/crds")"
 
 if ! git diff --quiet ${last_cilium_release}..${remote}/${release_version} $crd_path \
     && semverEQ "${current_release_version}" "${last_release_version}"; then

--- a/pkg/k8s/watchers/pod.go
+++ b/pkg/k8s/watchers/pod.go
@@ -1,4 +1,4 @@
-// Copyright 2016-2020 Authors of Cilium
+// Copyright 2016-2021 Authors of Cilium
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -80,6 +80,10 @@ func (k *K8sWatcher) createPodController(getter cache.Getter, fieldSelector fiel
 						} else {
 							metrics.EventLagK8s.Set(timeSinceEpCreated.Round(time.Second).Seconds())
 						}
+					} else {
+						// If the ep is nil then we reset to zero, otherwise
+						// the previous value set is kept forever.
+						metrics.EventLagK8s.Set(0)
 					}
 					err := k.addK8sPodV1(pod)
 					k.K8sEventProcessed(metricPod, metricCreate, err == nil)

